project/Build.scala

```diff
@@ -775,6 +775,19 @@ object Build {
           previousArtifactSetting,
           mimaBinaryIssueFilters ++= BinaryIncompatibilities.SbtPlugin,
 
+          /* Work around https://github.com/sbt/sbt/issues/3393.
+           * Should be:
+           *   addSbtPlugin("org.portable-scala" % "sbt-platform-deps" % "1.0.0-M2")
+           * We inline the fixed definition of addSbtPlugin to be
+           * released with sbt 0.13.17.
+           */
+          libraryDependencies += {
+            Defaults.sbtPluginExtra(
+                "org.portable-scala" % "sbt-platform-deps" % "1.0.0-M2",
```

Comment:
The 0.6 branch may not depend on a milestone build of sbt-platform-deps. This breaks our own binary compatibility guarantee of the sbt plugin, no? (Next release of 0.6 wouldn't be binary compatible with one that uses sbt-platform-deps 1.0.0).

Reply:
Ah, right.

On the other hand, I don't except sbt-platform-deps 1.0.0 to break binary compatibility wrt. 1.0.0-M2. In fact I don't expect sbt-platform-deps 1.0.0 to be any different than 1.0.0-M2.

Maybe we just publish sbt-platform-deps 1.0.0 final right now?

/cc @densh

Reply:
Or we let this PR dormant for a bit while we publish Scala.js 1.0.0-M2, itself depending on sbt-platform-deps 1.0.0-M2, to give it some exposure in the wild. And if we're happy with it, only then we publish 1.0.0 final and come back to the present PR.

Reply:
That sounds very reasonable.

Reply:
Actually, just to double check: That will not make 0.6 / 1.0 cross compilation harder, will It?

Reply:
No, I already checked that one can use `sbt-scalajs-crossproject` 0.3.0 with any of the following:

* Scala.js 0.6.19 (and hence, I kind of assumed, 0.6.21 or the current 0.6.x branch)
* Scala.js 0.6.22-SNAPSHOT from this branch
* Scala.js 1.0.0-M2-SNAPSHOT from the branch of #3197.

And all of that, with the exact same codebase (except the change of version number for `sbt-scalajs`).